### ISSUE TITLE
Userland working fine except for net speed

### ISSUE THREAD

**Issue by Sergpot** [CypherpunkArmory/UserLAnd#501]
First and foremost, I don't think the below issue relates to a bug, it seems like something that maybe can be fixed with some proper configuration tweaks, although I don't know which.
I'm running Userland with both Arch and Ubuntu inside of it on an Asus Zenpad 10 running Android 7.1 unrooted. Both distros work very fine except for the network speed which is drastically cut down in Userland (both in Arch and Ubuntu) with respect to the native Android data trasfer  rate (it is 1/10 and even less). In practice, the wlan and Internet connection are very very slow. As x server I use XSDL from the Play Store.
Any help would be greatly appreciated

**Comment by corbinlc** [comment]
Can you tell us how you tested the network speed?  Did you do this by installing packages via `apt` or by downloading a file via `wget` or something else?

**Comment by corbinlc** [comment]
Don't those results show fairly similar upload and download speeds for Android vs Debian (I assume that is a UserLAnd Debian filesystem)?

**Comment by EnderNightLord-ChromeBook** [comment]
No... It's the server / mirror if im in NZ and using a US mirror its gonna be slow, right? and if I use a NZ mirror then it's very quite fast.

**Comment by MatiasMFM2001** [comment]
I'm from Argentina. I made another test, now using a USA server (Thomasville) and the results are a bit less similar:

nPerf:

    Download: 10.34MBit/s max, 7.56MBit/s average
    Upload: 15.96MBit/s max, 14.14MBit/s average

speedtest-cli:

    Download: 5.59MBit/s (average?)
    Upload: 10.72MBit/s (average?)

**Comment by Sergpot** [comment]
Hello everybody,
I used speedtest-cli from the shell inside UserLand to make measurements. Outside of UserLand, i.e. in the hosting Android, I used a different tool (there's no speedtest-cli for Android). I made tests on three wlan networks (home ADSL, office wi-fi and mobile hotspot) always getting consistent poor results. What I haven't done still is testing against ethernet but I plan to do it later today; I shall report back to you after that. Another thing I plan to do is testing on Android for pc, which I've been using for some time now (Prime OS from XDA); also on this I will let you know soon.
KInd regards
(Sergio from Italy)


  

    Il lunedì 7 gennaio 2019, 21:02:59 CET, MatiasMFM2001 <notifications@github.com> ha scritto:  
 
 
I'm from Argentina. I made another test, now using a USA server (Thomasville) and the results are a bit less similar:

nPerf:
Download: 10.34MBit/s max, 7.56MBit/s average
Upload: 15.96MBit/s max, 14.14MBit/s average

speedtest-cli:
Download: 5.59MBit/s (average?)
Upload: 10.72MBit/s (average?)

**Comment by Sergpot** [comment]
Hello again,I managed to make some more tests at home today as planned. I installed UserLand and XSDL server apps on a pc running the latest Android Prime OS.I fumbled a little bit by varying the combination of Linux distro and DE, which took me some time but gave the opportunity to get acquainted with the operation and find the optimal association (BTW, in the end I got quite satisfied with Mate on top of Debian which I plan to keep). As I tried each combination distro/DE I also tested the network performance over my home ADSL and this time I got consistently comparable results, no material delay I could observe in UserLand. Good.That suggested me that either the Android tablet had a different operation or I had just made some mistake in the setup of the distro/DE combination. So I reinstalled everything and set up again all the stuff with much more attention and the experience gained on the pc this time.Now the tablet is working the same speed as the native Android, at least on my home network. Some measurements even gave me better figures!So in the end I was wrong in something, and that's all.I hope I didn't bug youSergioP.S. the very idea of UserLand, not to mention the brilliant implementation of it, it's just awesome to me. Should developer/s be reading this, be proud of it, it's fully deserved  


Inviato da Yahoo Mail su Android 
 
  Il mar, 8 gen, 2019 alle 9:12, Sergio Potenza<sergio.potenza@yahoo.it> ha scritto:   Hello everybody,
I used speedtest-cli from the shell inside UserLand to make measurements. Outside of UserLand, i.e. in the hosting Android, I used a different tool (there's no speedtest-cli for Android). I made tests on three wlan networks (home ADSL, office wi-fi and mobile hotspot) always getting consistent poor results. What I haven't done still is testing against ethernet but I plan to do it later today; I shall report back to you after that. Another thing I plan to do is testing on Android for pc, which I've been using for some time now (Prime OS from XDA); also on this I will let you know soon.
KInd regards
(Sergio from Italy)


  

    Il lunedì 7 gennaio 2019, 21:02:59 CET, MatiasMFM2001 <notifications@github.com> ha scritto:  
 
 
I'm from Argentina. I made another test, now using a USA server (Thomasville) and the results are a bit less similar:

nPerf:
Download: 10.34MBit/s max, 7.56MBit/s average
Upload: 15.96MBit/s max, 14.14MBit/s average

speedtest-cli:
Download: 5.59MBit/s (average?)
Upload: 10.72MBit/s (average?)
